[GPIO] Add checks for GPIO boot states

TD-er opened the pull request:
Added some checks for the GPIO boot states.
Changing boot state will now set the new state. (was only active after reboot)

TD-er committed:
[GPIO] Add checks for GPIO boot states

Added some checks for the GPIO boot states.
Changing boot state will now set the new state. (was only active after reboot)

TD-er committed:
[GPIO] Move portStatusStruct and related functions to separate file

And some clean-up of the code.

TD-er committed:
[CSS] Fix padding in menu.

`.menu {padding: 4px 16px 8px;}` should be `.menu {padding: 4px 16px 4px;}`

TD-er committed:
[GPIO] Simple clean-up to understand the portstatus code

See https://github.com/letscontrolit/ESPEasy/pull/2057#issuecomment-445247213

TD-er committed:
[GPIO] Use helper class for duplicate code in P001, P009 and P019

Moved duplicate code to functions used in:
- PLUGIN_WEBFORM_LOAD
- PLUGIN_WEBFORM_SAVE

TD-er committed:
[GPIO] Cleanup and documentation of GPIO code.

giig1967g reviewed (commented):
In function `readyToDelete` there is no check for init flag. Is it on purpose?

TD-er commented:
@giig1967g It is like it was implemented.
In one occasion it was not checking for init and so I didn't add it to that function.
In places where it was checking for init, I still do check for it.

TD-er committed:
[GPIO] Use descriptive get/set functions to access settings

For example:
```c++
int16_t hlp_getUseDoubleClick(byte taskIndex) { 
  return Settings.TaskDevicePluginConfig[taskIndex][4]; 
} 
 
void hlp_setUseDoubleClick(byte taskIndex, int16_t value) { 
   Settings.TaskDevicePluginConfig[taskIndex][4] = value; 
} 
```

N.B. A number of temporary variables like counters and whether a long press has fired appear to be stored in the settings too.
This is something still needed to be fixed.

TD-er committed:
[GPIO] Move duplicate code to switch helper file.
